python/gtsam/tests/test_ShonanAveraging.py

```diff
@@ -192,9 +192,8 @@ def test_constructorBetweenFactorPose2s(self) -> None:
         wRi_list = [result_values.atRot2(i) for i in range(num_images)]
         thetas_deg = np.array([wRi.degrees() for wRi in wRi_list])
 
-        # map all angles to [0,360)
-        thetas_deg = thetas_deg % 360
-        thetas_deg -= thetas_deg[0]
+        # map all angles to [-180,180)
+        thetas_deg = (thetas_deg - thetas_deg[0] + 180) % 360 - 180
```

Comment:
reasoning here?

Reply:
The existing angle normalization was a bit fragile and after the changes in this PR the computed value was `[0.0, -270.0, 0.0]` instead of the expected `[0.0, 90.0, 0.0]`. I assume that's an innocuous modulo error and nothing deeper.
I modified the logic to ensure that the final values are in the range [-180,180) instead of [0,360), so any slightly negative values are still close to zero.